### PR TITLE
CI: Change clang-format version to 11

### PR DESCRIPTION
**Pull request by maiadegraaf**
Brew only provides clang-format11 (which works).
The version that Python provides when I install it through pip (python3 -m pip install clang_format==10.0.1.1) can not be executed by my CPU:
```
OSError: [Errno 86] Bad CPU type in executable: '/opt/homebrew/lib/python3.11/site-packages/clang_format/data/bin/clang-format'
```

So we can't have this version requirement on 10, it might be a saner solution to move everything to clang-format 11 instead, as it is more readily available.

@Mause

### CODE REVIEW

File: scripts/format.py [hunk 1]
```diff
@@ -20,9 +20,6 @@
     exit(-1)
 
 ver = subprocess.check_output(('clang-format', '--version'), text=True)
-if '10.' not in ver:
```
Comment (Mause) [inline scripts/format.py:23]: I'd prefer we keep this code and update it to point to version 11 instead, if you can't install clang-format 11

Obviously that would require that we update the Code Format check run

**Review by Mause — CHANGES_REQUESTED** [review]

File: scripts/format.py (hunk 1 above)
Comment (maiadegraaf) [reply scripts/format.py:23]: Sure! I had a go at changing it myself, but not sure if I did it correctly.

**Comment by maiadegraaf** [comment]
Changed the clang-format version to 11 to prevent the `Bad CPU type in executable` error, and ran the `format.py` script with the new version.

**Comment by Mytherin** [comment]
Thanks! LGTM

**Pull request merged**